src/NHibernate/Multi/ICachingInformation.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections;
+using System.Collections.Generic;
+using NHibernate.Cache;
+using NHibernate.Engine;
+using NHibernate.Type;
+
+namespace NHibernate.Multi
+{
+	/// <summary>
+	/// Querying information.
+	/// </summary>
+	public interface ICachingInformation
+	{
+		/// <summary>
+		/// The query parameters.
+		/// </summary>
+		QueryParameters Parameters { get; }
+
+		/// <summary>
+		/// The query result.
+		/// </summary>
+		IList Result { get; }
```

Comment:
May be it should be exposed as `GetResultsToCache` or some other name that suggests that implementor should return values that's going to be put to cache.

Reply:
If doing so, it should then hold neither results retrieved from cache nor results after cacheable transformation which should not be put (the put must be done on the untransformed result).